### ISSUE TITLE
Replace "request.respondWith()" with a Controller

### ISSUE THREAD

**Issue by kettanaito** [mswjs/interceptors#431]
Instead of adding the `respondWith()` custom method on the Fetch API `Request` representation of the intercepted request, expose that method _alongside_ the request in the listener argument object.

```js
interceptor.on('request', (event) => {
  const { request } = event
  event.respondWith(new Response())
})
```

This also resembles the Service Worker's `fetch` event a bit more closely. Not that it has ever the intention but it's a nice little detail.

**Comment by kettanaito** [comment]
## Request controller interface

```ts
interface RequestController {
  // Instruct the interceptor to perform the request as-is.
  passthrough(): void

  // Use the given response as the mock response for the request.
  respondWith(response: Response): void

  // Use the given error as the request error.
  errorWith(error?: Error): void

  // Abort the request.
  // In some clients (ClientRequest) it's synonymous to ".errorWith()",
  // in others (XMLHttpRequest) it's different. 
  abort(): void
}
```

The request controller is exposed in the `request` listener argument:

```js
interceptor.on('request', ({ request, controller }) => {
  controller.respondWith(new Response('hello world'))
})
```

The request controller can only handle the request _once_. Once any of its methods are called, calling methods on it again must throw an error.

**Comment by kettanaito** [comment]
## Released: v0.33.0 🎉

This has been released in v0.33.0!

- 📄 [**Release notes**](https://github.com/mswjs/interceptors/releases/tag/v0.33.0)
- 📦 [npm package](https://www.npmjs.com/package/@mswjs/interceptors/v/0.33.0)

Make sure to always update to the latest version (`npm i @mswjs/interceptors@latest`) to get the newest features and bug fixes.

---

_Predictable release automation by [@ossjs/release](https://github.com/ossjs/release)_.
